src/update-done/update-done.c

```diff
@@ -36,6 +36,7 @@ static int apply_timestamp(const char *path, struct timespec *ts) {
         _cleanup_fclose_ FILE *f = NULL;
         int fd = -1;
         int r;
+        char *tmp;
```

Comment:
This leaks memory: nothing frees the "tmp" string here... Also, if we fail creating the timestamp properly for some reason we will leave the temporary file around if O_TMPFILE wasn't available. Thus, we really should remove the file on failure here. 

Doing both is pretty easy, just change the above to:

```
_cleanup_(unlink_and_freep) char *tmp = NULL;
```

This uses the "cleanup" gcc extension to unlink the file and free the string when it goes out of scope and is not NULL.